server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java

```diff
@@ -0,0 +1,214 @@
+            "time",
+            nodeStatsCache.indicesMergeTotalTimeInMillisSupplier()
+        );
+        registry.registerLongGauge(
+            "es.node.stats.indices.translog.operations",
+            "Number of transaction log operations.",
+            "operation",
+            nodeStatsCache.indicesTranslogOperationSupplier()
+        );
+        registry.registerLongGauge(
+            "es.node.stats.indices.translog.size",
+            "Size, in bytes, of the transaction log.",
+            "bytes",
+            nodeStatsCache.indicesTranslogSizeInBytesSupplier()
+        );
+        registry.registerLongGauge(
+            "es.node.stats.indices.translog.uncommitted_operations",
+            "Number of uncommitted transaction log operations.",
+            "operations",
+            nodeStatsCache.indicesTranslogUncommittedOperationsSupplier()
+        );
+        registry.registerLongGauge(
+            "es.node.stats.indices.translog.uncommitted_size_in_bytes",
+            "Size, in bytes, of uncommitted transaction log operations.",
+            "bytes",
+            nodeStatsCache.indicesTranslogUncommittedSizeInBytesSupplier()
+        );
+        registry.registerLongGauge(
+            "es.node.stats.indices.translog.earliest_last_modified_age",
+            "Earliest last modified age for the transaction log.",
+            "time",
+            nodeStatsCache.indicesTranslogEariestLastModifiedAgeSupplier()
+        );
+        registry.registerLongGauge(
+            "es.node.stats.http.current_open",
+            "Current number of open HTTP connections for the node.",
+            "connections",
+            nodeStatsCache.httpCurrentOpenSupplier()
+        );
+        registry.registerLongGauge(
+            "es.node.stats.transport.rx_size_in_bytes",
+            "Size, in bytes, of RX packets received by the node during internal cluster communication.",
+            "bytes",
+            nodeStatsCache.transportRxSizeInBytesSupplier()
+        );
+        registry.registerLongGauge(
+            "es.node.stats.transport.tx_size_in_bytes",
+            "Size, in bytes, of TX packets sent by the node during internal cluster communication.",
+            "bytes",
+            nodeStatsCache.transportTxSizeInBytesSupplier()
+        );
+    }
+
+    private NodeStats getNodeStats() {
+        CommonStatsFlags flags = new CommonStatsFlags(
+            CommonStatsFlags.Flag.Get,
+            CommonStatsFlags.Flag.Search,
+            CommonStatsFlags.Flag.Merge,
+            CommonStatsFlags.Flag.Translog
+        );
+        return nodeService.stats(
+            flags,
+            true,
+            false,
+            false,
+            false,
+            false,
+            false,
+            true,
+            true,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false
+        );
+    }
```

Comment:
Originally it was but since I am using it for setting the initialValue in the NodeStatsCache `super(...)` I get the following error: `Cannot reference 'NodeStatsCache.getNodeStats()' before supertype constructor has been called`. Therefore I needed either to duplicate the code or move it out and went for the second option.